test/refactoring/ConvertAsync/convert_function.swift

```diff
@@ -345,4 +366,26 @@ func withImplicitReturn(completionHandler: (String) -> Void) {
 // IMPLICIT-RETURN: func withImplicitReturn() async -> String {
 // IMPLICIT-RETURN-NEXT:   let val0 = await simple()
 // IMPLICIT-RETURN-NEXT:   return val0
-// IMPLICIT-RETURN-NEXT: }
\ No newline at end of file
+// IMPLICIT-RETURN-NEXT: }
+
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=ADD-ALWAYS-NIL-ERROR %s
+func addAlwaysNilError(completion: (String?, Error?) -> Void) {
+  simple { res in
+    completion(res, nil)
+  }
+}
+// ADD-ALWAYS-NIL-ERROR:      func addAlwaysNilError() async throws -> String {
+// ADD-ALWAYS-NIL-ERROR-NEXT:   let res = await simple()
+// ADD-ALWAYS-NIL-ERROR-NEXT:   return res
+// ADD-ALWAYS-NIL-ERROR-NEXT: }
+
+// RUN: %refactor -convert-to-async -dump-text -source-filename %s -pos=%(line+1):1 | %FileCheck -check-prefix=RELAY-ERROR-AND-RESULT %s
+func relayErrorAndResult(completion: (String?, Error?) -> Void) {
+  simpleErr(arg: "test") { (res, err) in
+    completion(res, err)
+  }
+}
+// RELAY-ERROR-AND-RESULT:      func relayErrorAndResult() async throws -> String {
+// RELAY-ERROR-AND-RESULT-NEXT:   let res = try await simpleErr(arg: "test")
+// RELAY-ERROR-AND-RESULT-NEXT:   return res
+// RELAY-ERROR-AND-RESULT-NEXT: }
```

Comment:
Add a test for non-nil error here? Ie.
```
simple { res in
  completion(res, MyError())
}
```

Reply:
Added one and realized that we were actually dropping the error there (because we had an ambiguous call inside the success block). I changed the heuristic to 
- only interpret an ambiguous call as a success call if it isn the success branch and the error has been placeholdered
- add `assert`s that the success arguments are `nil` if we continue to interpret the ambiguous call as an error call